examples/calendar/backend/services/time/controller.ts

```diff
@@ -0,0 +1,92 @@
+import { NextFunction, Request, Response } from "express";
+//import { User } from "../../models/user.model";
+import { Time } from "../../models/time.model";
+import { Op } from "sequelize";
+
+// Create time availablity for the user
+export const create = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if ((req as any).user.payload.id !== +req.body.userId) {
+    return res.status(401).send({ error: "You can can only access yourself" });
+  }
+  const { timeStarts, timeEnds, userId } = req.body;
+  if (timeStarts.length !== timeEnds.length) {
+    return res
+      .status(400)
+      .send({ error: "Something went wrong! Please contact developer" });
+  }
+  try {
+    const times = [];
+    for (let i = 0; i < timeStarts.length; i++) {
+      times.push({
+        timestart: timeStarts[i],
+        timeend: timeEnds[i],
+        userId: userId,
+        status: "available",
+      });
+    }
+    const timejson = await Time.bulkCreate(times);
+    return res.status(200).send(timejson);
+  } catch (error) {
+    console.log(error);
+    return res.status(400);
+  }
+};
+
+// update time availability for the user.
+export const update = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if ((req as any).user.payload.id !== +req.params.userId) {
+    return res.status(401).send({ error: "You can can only access yourself" });
+  }
+  const { ids, timeStarts, timeEnds } = req.body;
+  if (timeStarts.length !== timeEnds.length) {
+    return res
+      .status(400)
+      .send({ error: "Something went wrong! Please contact developer" });
```

Comment:
please refer the previous comment in this file

Reply:
I have updated this.